api/client/teleport_resolver.go

```diff
@@ -0,0 +1,174 @@
+/*
```

Comment:
Replace resolver with creating a stream to the auth server to know when service configuration changes.

This is going to require a new load balancer that is in between the default pick_first load balancer. Should be able to switch back and forth between pick_first and teleport_pick_healthy.